data/shared/citizen/scripting/v8/timer.js

```diff
@@ -1,62 +1,102 @@
 // Timers
 
-(function (global) {
+(function(global) {
     let gameTime = Citizen.getTickCount();
 
     const timers = {};
     let timerId = 0;
-    const tickers = [];
+
+    const tickers = {};
+    let tickerId = 0;
+
     let animationFrames = [];
 
-    function  setTimer(timer, callback, interval) {
+    function printError(where, e) {
+        console.error(`Unhandled error in ${where}: ${e.toString()}\n${e.stack}`);
+    }
+
+    function setTimer(timer, callback, interval) {
         timers[timer.id] = {
             callback,
             interval,
             lastRun: gameTime
         };
     }
 
-    function nextId() {
-        return { id: timerId++, unref() {}, ref() {} };
+    function nextTimerId() {
+        return {
+            id: ++timerId,
+            unref() {},
+            ref() {}
+        };
+    }
+
+    function nextTickerId() {
+        return ++tickerId;
     }
 
     function setTick(callback) {
-        tickers[tickers.length] = callback;
+        const id = nextTickerId();
+
+        tickers[id] = {
+            callback,
+            promise: null,
+        };
+
+        return id;
     }
 
-    function clearTimer(timer) {
-        if (!timer) { return; }
-    
-        delete timers[timer.id];
+    function clearTick(ticker) {
+        if (!ticker) {
+            return;
+        }
+
+        delete tickers[ticker];
     }
 
-    function setTick(callback) {
-        tickers[tickers.length] = callback;
+    function resolveTicker(ticker) {
```

Comment:
On L151, you're binding `tickerId` [to the context] - not as an argument...

You probably need to replace `ticker` (`tickers[ticker]`) with `this`

Reply:
You're right, thanks for pointing this one out!